<a href="https://colab.research.google.com/github/EmamulHossen/Feature-Transformation-Assignment-6.3-/blob/main/FT(Assignment_6_3).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive

In [2]:
drive.mount("/content/drive")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# **Import Necessary Library**

In [99]:
import warnings
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt


warnings.filterwarnings("ignore")
%matplotlib inline

#**Part-1 : Data Preprocessing**

## **Load Dataset**

In [4]:
df=pd.read_csv("/content/drive/MyDrive/Data/Real estate.csv")

In [5]:
df.head()

No  X1 transaction date  X2 house age  \
0   1             2012.917          32.0   
1   2             2012.917          19.5   
2   3             2013.583          13.3   
3   4             2013.500          13.3   
4   5             2012.833           5.0   

   X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                84.87882                               10   
1                               306.59470                                9   
2                               561.98450                                5   
3                               561.98450                                5   
4                               390.56840                                5   

   X5 latitude  X6 longitude  Y house price of unit area  
0     24.98298     121.54024                        37.9  
1     24.98034     121.53951                        42.2  
2     24.98746     121.54391                        47.3  
3     24.98746     121.54391                        54.8  
4     24.97937     121.54245                        43.1

In [6]:
#Check the shape of dataset
df.shape

(414, 8)

In [7]:
#Check the shape of dataset
print("Shape of the dataset: {}".format(df.shape))
print("Shape of dataset: Number of Row:{},Number of Column:{}".format(df.shape[0],df.shape[1]))

Shape of the dataset: (414, 8)
Shape of dataset: Number of Row:414,Number of Column:8


**Checking the null/ missing value**

In [8]:
missing_value=df.isnull().any()

In [9]:
missing_value

No                                        False
X1 transaction date                       False
X2 house age                              False
X3 distance to the nearest MRT station    False
X4 number of convenience stores           False
X5 latitude                               False
X6 longitude                              False
Y house price of unit area                False
dtype: bool

In [10]:
missing_value=df.isnull().sum()

In [11]:
missing_value

No                                        0
X1 transaction date                       0
X2 house age                              0
X3 distance to the nearest MRT station    0
X4 number of convenience stores           0
X5 latitude                               0
X6 longitude                              0
Y house price of unit area                0
dtype: int64

**Extract the feature column(x1 to x6)**

In [12]:
x=df.drop(columns=["No","Y house price of unit area"],axis=1)

In [13]:
x.head() # Show the first five row

X1 transaction date  X2 house age  X3 distance to the nearest MRT station  \
0             2012.917          32.0                                84.87882   
1             2012.917          19.5                               306.59470   
2             2013.583          13.3                               561.98450   
3             2013.500          13.3                               561.98450   
4             2012.833           5.0                               390.56840   

   X4 number of convenience stores  X5 latitude  X6 longitude  
0                               10     24.98298     121.54024  
1                                9     24.98034     121.53951  
2                                5     24.98746     121.54391  
3                                5     24.98746     121.54391  
4                                5     24.97937     121.54245

In [14]:
x.tail() # Showing the last five row used tail function

X1 transaction date  X2 house age  \
409             2013.000          13.7   
410             2012.667           5.6   
411             2013.250          18.8   
412             2013.000           8.1   
413             2013.500           6.5   

     X3 distance to the nearest MRT station  X4 number of convenience stores  \
409                              4082.01500                                0   
410                                90.45606                                9   
411                               390.96960                                7   
412                               104.81010                                5   
413                                90.45606                                9   

     X5 latitude  X6 longitude  
409     24.94155     121.50381  
410     24.97433     121.54310  
411     24.97923     121.53986  
412     24.96674     121.54067  
413     24.97433     121.54310

In [15]:
y=df["Y house price of unit area"]

In [16]:
y.head()

0    37.9
1    42.2
2    47.3
3    54.8
4    43.1
Name: Y house price of unit area, dtype: float64

**Split the dataset into features(x) as 70% traing and 30% testing ratio**

In [17]:
from sklearn.model_selection import train_test_split

In [18]:
X_train, X_test, y_train, y_test = train_test_split( x, y, test_size=0.30, random_state=42)

In [19]:
print("Shape of X_train:",X_train.shape)
print("Shape of X_test :",X_test.shape)
print("Shape of y_train",y_train.shape)
print("Shape of y_test:",y_test.shape)

Shape of X_train: (289, 6)
Shape of X_test : (125, 6)
Shape of y_train (289,)
Shape of y_test: (125,)


# **Linear Regression**
>**Without Feature Scaling**



In [20]:
from sklearn.linear_model import LinearRegression

In [21]:
LR_model=LinearRegression()

In [22]:
LR_model.fit(X_train,y_train)

LinearRegression()

**Coefficients**

In [23]:
#Slope
m_=LR_model.coef_

In [24]:
m_

array([ 5.84779706e+00, -2.42545813e-01, -5.13873381e-03,  1.07449530e+00,
        2.39096949e+02, -5.22351591e+01])

In [25]:
#Intercept
c_=LR_model.intercept_

In [26]:
c_

-11350.740518771527

In [27]:
y_train_predict=LR_model.predict(X_train)

In [28]:
y_train_predict

array([47.32074756, 15.10754801, 46.95629589, 11.8601291 , 30.6116461 ,
       35.23397379, 40.42310691, 45.50708821, 40.18349801, 36.83495501,
       34.09696749, 37.70337476, 27.79080015, 34.70049596, 24.55194289,
       37.13787565, 44.68117429, 52.45976408, 37.23468772, 47.94245238,
       41.42892542, 42.58464002, 43.23127535, 45.21550123, 31.44630639,
       44.81269688, 30.47900933, 39.20917476, 42.19179401, 54.35199522,
       44.25551663, 34.41635164, 40.73665437, 45.86719965, 43.01690126,
       39.93460642, 38.88477727, 48.29378192, 37.07582259, 45.12882333,
       41.07510261, 34.66401256, 12.08437271, 43.15586926, 38.14849191,
       43.15586926, 51.38405277, 45.93279355, 34.48071984, 38.91388548,
       27.84017764, 51.68302612, 43.89392614, 16.18414497, 41.27000829,
       47.34623383, 46.08764064, 43.82061184, 28.00014687, 44.91111927,
       44.64675952, 53.42977007, 43.91060764, 44.86743814, 38.26205457,
        6.172469  , 43.99788821, 47.56691057, 29.88167717, 29.41

In [29]:
y_train

7       46.7
194     29.3
19      47.7
176     19.2
367     20.9
       ...  
71      40.8
106     47.1
270    117.5
348     53.7
102     54.4
Name: Y house price of unit area, Length: 289, dtype: float64

In [30]:
y_test_predict=LR_model.predict(X_test)

In [31]:
y_test_predict

array([47.55430212, 41.08372744, 44.25551663, 40.51685112, 27.43467608,
       45.85435624, 44.70296876, 46.57752144, 23.86072119, 54.35199522,
       34.63011363, 36.34958785, 37.34518438, 23.49428018, 36.02735916,
       34.56828629, 44.01823263, 45.04246804, 29.9554971 , 43.96338099,
        0.09450675, 35.291724  , 48.09455663, 45.54503941, 10.6006418 ,
       39.9786752 , 11.9116313 , 42.79356737, 35.67204264, 38.09573519,
       12.98819108, 40.13164278, 35.44632378, 27.09263458, 46.57230583,
       32.08134705, 50.62714445, 13.2533591 , 49.8007962 , 40.83574406,
       39.02898306, 40.92260381, 47.61743515, 37.68719274, 39.67028734,
       46.64468542, 43.57375428, 19.40851794, 46.95629589, 45.93056809,
       48.03966928, 48.79469044, 43.06419598, 43.00021216, 36.56049741,
       14.20568662, 37.31495695, 34.30916207, 28.33264665, 46.09215428,
       32.92580137, 32.83028385, 14.20568662,  8.3906898 ,  9.79703308,
       34.33939647, 27.76485421, 46.75860125, 33.77722462, 30.85

In [32]:
y_test

358    45.1
350    42.3
373    52.2
399    37.3
369    22.8
       ... 
268    40.1
148    45.1
16     70.1
66     50.7
341    37.0
Name: Y house price of unit area, Length: 125, dtype: float64

In [33]:
#Accuracy
LR_model_accuracy=LR_model.score(X_test,y_test)

In [34]:
LR_model_accuracy

0.5600638168604195

**Feature Scaling**

In [35]:
for i in x:
    mean = x[i].mean()
    std = x[i].std()
    x[i] = (x[i] - mean) / std

x.head()

X1 transaction date  X2 house age  X3 distance to the nearest MRT station  \
0            -0.822688      1.254111                               -0.791537   
1            -0.822688      0.156896                               -0.615866   
2             1.539289     -0.387322                               -0.413515   
3             1.244928     -0.387322                               -0.413515   
4            -1.120595     -1.115873                               -0.549332   

   X4 number of convenience stores  X5 latitude  X6 longitude  
0                         2.004982     1.124070      0.448220  
1                         1.665488     0.911342      0.400654  
2                         0.307513     1.485063      0.687352  
3                         0.307513     1.485063      0.687352  
4                         0.307513     0.833180      0.592220

In [36]:
x

X1 transaction date  X2 house age  \
0              -0.822688      1.254111   
1              -0.822688      0.156896   
2               1.539289     -0.387322   
3               1.244928     -0.387322   
4              -1.120595     -1.115873   
..                   ...           ...   
409            -0.528327     -0.352211   
410            -1.709316     -1.063206   
411             0.358300      0.095452   
412            -0.528327     -0.843763   
413             1.244928     -0.984207   

     X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                 -0.791537                         2.004982   
1                                 -0.615866                         1.665488   
2                                 -0.413515                         0.307513   
3                                 -0.413515                         0.307513   
4                                 -0.549332                         0.307513   
..                                      ...                              ...   
409                                2.375490                        -1.389957   
410                               -0.787118                         1.665488   
411                               -0.549014                         0.986500   
412                               -0.775745                         0.307513   
413                               -0.787118                         1.665488   

     X5 latitude  X6 longitude  
0       1.124070      0.448220  
1       0.911342      0.400654  
2       1.485063      0.687352  
3       1.485063      0.687352  
4       0.833180      0.592220  
..           ...           ...  
409    -2.214314     -1.925506  
410     0.427062      0.634573  
411     0.821899      0.423460  
412    -0.184532      0.476238  
413     0.427062      0.634573  

[414 rows x 6 columns]

In [37]:
x1=x.copy()

In [38]:
x1

X1 transaction date  X2 house age  \
0              -0.822688      1.254111   
1              -0.822688      0.156896   
2               1.539289     -0.387322   
3               1.244928     -0.387322   
4              -1.120595     -1.115873   
..                   ...           ...   
409            -0.528327     -0.352211   
410            -1.709316     -1.063206   
411             0.358300      0.095452   
412            -0.528327     -0.843763   
413             1.244928     -0.984207   

     X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                 -0.791537                         2.004982   
1                                 -0.615866                         1.665488   
2                                 -0.413515                         0.307513   
3                                 -0.413515                         0.307513   
4                                 -0.549332                         0.307513   
..                                      ...                              ...   
409                                2.375490                        -1.389957   
410                               -0.787118                         1.665488   
411                               -0.549014                         0.986500   
412                               -0.775745                         0.307513   
413                               -0.787118                         1.665488   

     X5 latitude  X6 longitude  
0       1.124070      0.448220  
1       0.911342      0.400654  
2       1.485063      0.687352  
3       1.485063      0.687352  
4       0.833180      0.592220  
..           ...           ...  
409    -2.214314     -1.925506  
410     0.427062      0.634573  
411     0.821899      0.423460  
412    -0.184532      0.476238  
413     0.427062      0.634573  

[414 rows x 6 columns]

In [39]:
from sklearn.model_selection import train_test_split

In [40]:
X1_train, X1_test, y_train, y_test = train_test_split( x1, y, test_size=0.30, random_state=42)

# **Part-2 : Linear Regression**
---
**Using Scaled Value**


In [41]:
from sklearn.linear_model import LinearRegression

In [42]:
model=LinearRegression()

In [43]:
model.fit(X1_train,y_train)

LinearRegression()

**Regression Coefficients**

In [44]:
#Slope
M=model.coef_


In [45]:
M

array([ 1.6488872 , -2.76319942, -6.48564526,  3.16499231,  2.96724014,
       -0.80166255])

In [46]:
#Intercept
C=model.intercept_

In [47]:
C

37.983867630237256

In [48]:
#prediction
y_predict_train=model.predict(X1_train)

In [49]:
y_predict_train

array([47.32074756, 15.10754801, 46.95629589, 11.8601291 , 30.6116461 ,
       35.23397379, 40.42310691, 45.50708821, 40.18349801, 36.83495501,
       34.09696749, 37.70337476, 27.79080015, 34.70049596, 24.55194289,
       37.13787565, 44.68117429, 52.45976408, 37.23468772, 47.94245238,
       41.42892542, 42.58464002, 43.23127535, 45.21550123, 31.44630639,
       44.81269688, 30.47900933, 39.20917476, 42.19179401, 54.35199522,
       44.25551663, 34.41635164, 40.73665437, 45.86719965, 43.01690126,
       39.93460642, 38.88477727, 48.29378192, 37.07582259, 45.12882333,
       41.07510261, 34.66401256, 12.08437271, 43.15586926, 38.14849191,
       43.15586926, 51.38405277, 45.93279355, 34.48071984, 38.91388548,
       27.84017764, 51.68302612, 43.89392614, 16.18414497, 41.27000829,
       47.34623383, 46.08764064, 43.82061184, 28.00014687, 44.91111927,
       44.64675952, 53.42977007, 43.91060764, 44.86743814, 38.26205457,
        6.172469  , 43.99788821, 47.56691057, 29.88167717, 29.41

In [50]:
y_train

7       46.7
194     29.3
19      47.7
176     19.2
367     20.9
       ...  
71      40.8
106     47.1
270    117.5
348     53.7
102     54.4
Name: Y house price of unit area, Length: 289, dtype: float64

In [51]:
y_prediction_test=model.predict(X1_test)

In [52]:
y_prediction_test

array([47.55430212, 41.08372744, 44.25551663, 40.51685112, 27.43467608,
       45.85435624, 44.70296876, 46.57752144, 23.86072119, 54.35199522,
       34.63011363, 36.34958785, 37.34518438, 23.49428018, 36.02735916,
       34.56828629, 44.01823263, 45.04246804, 29.9554971 , 43.96338099,
        0.09450675, 35.291724  , 48.09455663, 45.54503941, 10.6006418 ,
       39.9786752 , 11.9116313 , 42.79356737, 35.67204264, 38.09573519,
       12.98819108, 40.13164278, 35.44632378, 27.09263458, 46.57230583,
       32.08134705, 50.62714445, 13.2533591 , 49.8007962 , 40.83574406,
       39.02898306, 40.92260381, 47.61743515, 37.68719274, 39.67028734,
       46.64468542, 43.57375428, 19.40851794, 46.95629589, 45.93056809,
       48.03966928, 48.79469044, 43.06419598, 43.00021216, 36.56049741,
       14.20568662, 37.31495695, 34.30916207, 28.33264665, 46.09215428,
       32.92580137, 32.83028385, 14.20568662,  8.3906898 ,  9.79703308,
       34.33939647, 27.76485421, 46.75860125, 33.77722462, 30.85

In [53]:
y_test

358    45.1
350    42.3
373    52.2
399    37.3
369    22.8
       ... 
268    40.1
148    45.1
16     70.1
66     50.7
341    37.0
Name: Y house price of unit area, Length: 125, dtype: float64

In [54]:
#Accuracy
Model_accuracy=model.score(X1_test,y_test)

In [55]:
Model_accuracy

0.5600638168604302

**Mean Squared Error(MSE)**
---
**Mean absolute Error(MAE)**


In [56]:
from sklearn.metrics import mean_squared_error,mean_absolute_error

In [57]:
MSE=mean_squared_error(y_test,y_prediction_test)

In [58]:
MSE

73.56837932850162

In [59]:
MAE=mean_absolute_error(y_test,y_prediction_test)

In [60]:
MAE

6.184836340096856

**R-Squred**

In [61]:
from sklearn.metrics import r2_score


In [62]:
r=r2_score(y_test,y_prediction_test)

In [63]:
r

0.5600638168604302

# **Part-3\Feature Scaling Comparision**

> **The prediction accuracy of Linear Regression for both (scaling data & Without scaling data) is same,which is 56%**



# **Part-4\Visualization & Analysis**

# **Part-5\Advanced Problem(optional)**

In [64]:
#Load Dataset
dataset=pd.read_csv("/content/drive/MyDrive/Data/Real estate.csv")

In [65]:
dataset.head()

No  X1 transaction date  X2 house age  \
0   1             2012.917          32.0   
1   2             2012.917          19.5   
2   3             2013.583          13.3   
3   4             2013.500          13.3   
4   5             2012.833           5.0   

   X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                84.87882                               10   
1                               306.59470                                9   
2                               561.98450                                5   
3                               561.98450                                5   
4                               390.56840                                5   

   X5 latitude  X6 longitude  Y house price of unit area  
0     24.98298     121.54024                        37.9  
1     24.98034     121.53951                        42.2  
2     24.98746     121.54391                        47.3  
3     24.98746     121.54391                        54.8  
4     24.97937     121.54245                        43.1

In [66]:
x5=dataset.drop(columns=["No","Y house price of unit area"],axis=1)

In [67]:
x5.head(10)

X1 transaction date  X2 house age  X3 distance to the nearest MRT station  \
0             2012.917          32.0                                84.87882   
1             2012.917          19.5                               306.59470   
2             2013.583          13.3                               561.98450   
3             2013.500          13.3                               561.98450   
4             2012.833           5.0                               390.56840   
5             2012.667           7.1                              2175.03000   
6             2012.667          34.5                               623.47310   
7             2013.417          20.3                               287.60250   
8             2013.500          31.7                              5512.03800   
9             2013.417          17.9                              1783.18000   

   X4 number of convenience stores  X5 latitude  X6 longitude  
0                               10     24.98298     121.54024  
1                                9     24.98034     121.53951  
2                                5     24.98746     121.54391  
3                                5     24.98746     121.54391  
4                                5     24.97937     121.54245  
5                                3     24.96305     121.51254  
6                                7     24.97933     121.53642  
7                                6     24.98042     121.54228  
8                                1     24.95095     121.48458  
9                                3     24.96731     121.51486

In [68]:
y5=dataset["Y house price of unit area"]

In [69]:
y5

0      37.9
1      42.2
2      47.3
3      54.8
4      43.1
       ... 
409    15.4
410    50.0
411    40.6
412    52.5
413    63.9
Name: Y house price of unit area, Length: 414, dtype: float64

In [70]:
from sklearn.preprocessing import MinMaxScaler

In [71]:
scaler=MinMaxScaler()

In [72]:
col_normalize=["X1 transaction date","X2 house age","X3 distance to the nearest MRT station","X4 number of convenience stores","X5 latitude","X6 longitude"]

In [73]:
x5[col_normalize]=scaler.fit_transform(x5[col_normalize])

In [74]:
x5[col_normalize]

X1 transaction date  X2 house age  \
0               0.272926      0.730594   
1               0.272926      0.445205   
2               1.000000      0.303653   
3               0.909389      0.303653   
4               0.181223      0.114155   
..                   ...           ...   
409             0.363537      0.312785   
410             0.000000      0.127854   
411             0.636463      0.429224   
412             0.363537      0.184932   
413             0.909389      0.148402   

     X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                  0.009513                              1.0   
1                                  0.043809                              0.9   
2                                  0.083315                              0.5   
3                                  0.083315                              0.5   
4                                  0.056799                              0.5   
..                                      ...                              ...   
409                                0.627820                              0.0   
410                                0.010375                              0.9   
411                                0.056861                              0.7   
412                                0.012596                              0.5   
413                                0.010375                              0.9   

     X5 latitude  X6 longitude  
0       0.616941      0.719323  
1       0.584949      0.711451  
2       0.671231      0.758896  
3       0.671231      0.758896  
4       0.573194      0.743153  
..           ...           ...  
409     0.114881      0.326504  
410     0.512118      0.750162  
411     0.571498      0.715225  
412     0.420141      0.723959  
413     0.512118      0.750162  

[414 rows x 6 columns]

In [75]:
from sklearn.model_selection import train_test_split

In [76]:
x5_train,x5_test,y_train,y_test=train_test_split(x5,y,test_size=0.3,random_state=42)

In [77]:
print("Shape of x5_train:",x5_train.shape)
print("Shape of x5_test :",x5_test.shape)
print("Shape of y_train",y_train.shape)
print("Shape of y_test:",y_test.shape)

Shape of x5_train: (289, 6)
Shape of x5_test : (125, 6)
Shape of y_train (289,)
Shape of y_test: (125,)


In [78]:
from sklearn.linear_model import LinearRegression

In [79]:
model_5=LinearRegression()

In [80]:
model_5.fit(x5_train,y_train)

LinearRegression()

In [81]:
y5_train_prediction=model_5.predict(x5_train)

In [82]:
y5_train_prediction

array([47.32074756, 15.10754801, 46.95629589, 11.8601291 , 30.6116461 ,
       35.23397379, 40.42310691, 45.50708821, 40.18349801, 36.83495501,
       34.09696749, 37.70337476, 27.79080015, 34.70049596, 24.55194289,
       37.13787565, 44.68117429, 52.45976408, 37.23468772, 47.94245238,
       41.42892542, 42.58464002, 43.23127535, 45.21550123, 31.44630639,
       44.81269688, 30.47900933, 39.20917476, 42.19179401, 54.35199522,
       44.25551663, 34.41635164, 40.73665437, 45.86719965, 43.01690126,
       39.93460642, 38.88477727, 48.29378192, 37.07582259, 45.12882333,
       41.07510261, 34.66401256, 12.08437271, 43.15586926, 38.14849191,
       43.15586926, 51.38405277, 45.93279355, 34.48071984, 38.91388548,
       27.84017764, 51.68302612, 43.89392614, 16.18414497, 41.27000829,
       47.34623383, 46.08764064, 43.82061184, 28.00014687, 44.91111927,
       44.64675952, 53.42977007, 43.91060764, 44.86743814, 38.26205457,
        6.172469  , 43.99788821, 47.56691057, 29.88167717, 29.41

In [83]:
y_train

7       46.7
194     29.3
19      47.7
176     19.2
367     20.9
       ...  
71      40.8
106     47.1
270    117.5
348     53.7
102     54.4
Name: Y house price of unit area, Length: 289, dtype: float64

In [84]:
y5_test_prediction=model_5.predict(x5_test)

In [85]:
y5_test_prediction

array([47.55430212, 41.08372744, 44.25551663, 40.51685112, 27.43467608,
       45.85435624, 44.70296876, 46.57752144, 23.86072119, 54.35199522,
       34.63011363, 36.34958785, 37.34518438, 23.49428018, 36.02735916,
       34.56828629, 44.01823263, 45.04246804, 29.9554971 , 43.96338099,
        0.09450675, 35.291724  , 48.09455663, 45.54503941, 10.6006418 ,
       39.9786752 , 11.9116313 , 42.79356737, 35.67204264, 38.09573519,
       12.98819108, 40.13164278, 35.44632378, 27.09263458, 46.57230583,
       32.08134705, 50.62714445, 13.2533591 , 49.8007962 , 40.83574406,
       39.02898306, 40.92260381, 47.61743515, 37.68719274, 39.67028734,
       46.64468542, 43.57375428, 19.40851794, 46.95629589, 45.93056809,
       48.03966928, 48.79469044, 43.06419598, 43.00021216, 36.56049741,
       14.20568662, 37.31495695, 34.30916207, 28.33264665, 46.09215428,
       32.92580137, 32.83028385, 14.20568662,  8.3906898 ,  9.79703308,
       34.33939647, 27.76485421, 46.75860125, 33.77722462, 30.85

In [86]:
y_test

358    45.1
350    42.3
373    52.2
399    37.3
369    22.8
       ... 
268    40.1
148    45.1
16     70.1
66     50.7
341    37.0
Name: Y house price of unit area, Length: 125, dtype: float64

In [87]:
#Accuracy
accuracy=model_5.score(x5_test,y_test)

In [88]:
accuracy

0.5600638168604036

In [89]:
from sklearn.metrics import mean_squared_error, mean_absolute_error
MSE_5 = mean_squared_error(y_test,y5_test_prediction)


In [90]:
MSE_5

73.56837932850604

In [91]:
model_5.intercept_

34.80019216997893

In [92]:
model_5.coef_

array([  5.3565821 , -10.6235066 , -33.22005471,  10.74495298,
        19.73028024,  -4.84428865])

In [93]:
from sklearn.metrics import r2_score

r_square = r2_score(y_test, y5_test_prediction)

In [94]:
print(r_square)

0.5600638168604036


# **Summary**
> **The performance of model has not affected by the feature scaling value.The performanc accuracy  both are same  which is 56%. But the value of intercept has occured variation on the basis of feature scaling value or not feature scaling value**

1.**The value of intercept(without feature scaling) is -11350.740518771527**<br/>
2.**The value of intercept(with feature scaling)is 37.983867630237256**


# **For advanced problem**
> **The performance accuracy is the same as above model.In this problem using scaling value ,the process of scaling MinMaxScaler.The value of intercept is 34.80019216997893**

